examples/Planning/ForecastReach.cs

```diff
@@ -0,0 +1,313 @@
+        {
+            GenerateReachForecastResponse response = reachPlanService.GenerateReachForecast(request);
+            Console.WriteLine("Reach curve output:");
+            Console.WriteLine(
+                "Currency,\tCost,\tOn-Target Reach,\tOn-Target Imprs,\tTotal Reach,\tTotal Imprs,\tProducts");
+            foreach (ReachForecast point in response.ReachCurve.ReachForecasts)
+            {
+                Console.Write($"{request.CurrencyCode},\t");
+                Console.Write($"{point.CostMicros / 1.0e6},\t");
+                Console.Write($"{point.Forecast.OnTargetReach},\t");
+                Console.Write($"{point.Forecast.OnTargetImpressions},\t");
+                Console.Write($"{point.Forecast.TotalReach},\t");
+                Console.Write($"{point.Forecast.TotalImpressions},\t");
+                Console.Write($"\"[");
+                foreach (ProductAllocation product in point.ForecastedProductAllocations)
+                {
+                    Console.Write($"(Product: {product.PlannableProductCode}, ");
+                    Console.Write($"Budget: {product.BudgetMicros / 1.0e6}), ");
+                }
+                Console.WriteLine($"]\"");
+            }
+        }
+
+        /// <summary>
+        /// Pulls a forecast for a manually allocated product mix.
+        /// </summary>
+        /// <param name="reachPlanService">Instance of Reach Plan Service client.</param>
+        /// <param name="customerId">The customer ID for the reach forecast.</param>
+        /// <param name="locationId">Location ID from somewhere returned by ListPlannableLocations.</param>
+        /// <param name="currencyCode">Three-character ISO 4217 currency code.</param>
+        /// <param name="budget">Budget in currency to plan for.</param>
+        public void ForecastManualMix(
+                        ReachPlanServiceClient reachPlanService,
+                        string customerId,
+                        string locationId,
+                        string currencyCode,
+                        long budget)
+        {
+            List<PlannedProduct> productMix = new List<PlannedProduct>();
+            productMix.Add(new PlannedProduct()
+            {
+                PlannableProductCode = "TRUEVIEW_IN_STREAM",
+                BudgetMicros = Convert.ToInt64((double)budget * 1.0e6 * 0.15)
+            });
+            productMix.Add(new PlannedProduct()
+            {
+                PlannableProductCode = "BUMPER",
+                BudgetMicros = Convert.ToInt64((double)budget * 1.0e6 * 0.85)
+            });
+
+            GenerateReachForecastRequest request =
+                BuildReachRequest(customerId, productMix, locationId, currencyCode);
+
+            PullReachCurve(reachPlanService, request);
+        }
+
+        /// <summary>
+        /// Pulls a forecast for a product mix suggested based on a set of preferences.
```

Comment:
What kind of preferences?

Reply:
Gave details on each kind of preference we support.